File: src/8장/8-2.py
import sys
from collections import deque
n = int(sys.stdin.readline())
queue = deque([])

for i in range(n):
    command = sys.stdin.readline().split()
    if command[0] == 'push':
        queue.append(command[1])
    elif command[0] == 'pop':
        if not queue:
            print(-1)
        else:
            print(queue.popleft())
    elif command[0] == 'size':
        print(len(queue))
    elif command[0] == 'empty':
        if not queue:
            print(1)
        else:
            print(0)
    elif command[0] == 'front':
        if not queue:
            print(-1)
        else:
            print(queue[0])
    elif command[0] == 'back':
        if not queue:
            print(-1)
        else:
            print(queue[-1])

File: src/8장/8-2.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Queue<String> queue = new LinkedList<>();
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int n = Integer.parseInt(bf.readLine());
		String back = "";
		for (int i = 0; i < n; i++) {
			String[] command = bf.readLine().split(" ");
			if (command[0].equals("push")) {
				back = command[1];
				queue.add(command[1]);
			} else if (command[0].equals("pop")) {
				bw.write(queue.isEmpty() ? "-1" : queue.poll());
				bw.newLine();
			} else if (command[0].equals("size")) {
				bw.write(String.valueOf(queue.size()));
				bw.newLine();
			} else if (command[0].equals("empty")) {
				bw.write(queue.isEmpty() ? "1" : "0");
				bw.newLine();
			} else if (command[0].equals("front")) {
				bw.write(queue.isEmpty() ? "-1" : queue.peek());
				bw.newLine();
			} else if (command[0].equals("back")) {
				bw.write(queue.isEmpty() ? "-1" : back);
				bw.newLine();
			}
		}
		bw.flush();
		bw.close();

	}
}